app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksViewModel.kt

```diff
@@ -142,19 +138,89 @@ class BookmarksViewModel @Inject constructor(
                 }
             }
         }
-        delete(savedSite)
+        hide(savedSite)
     }
 
-    private fun delete(savedSite: SavedSite) {
+    private fun hide(savedSite: SavedSite) {
+        // we don't delete, just remove from the ui list
+        when (savedSite) {
+            is Bookmark -> {
+                val bookmarks = viewState.value?.bookmarks!!.toMutableList()
+                val index = bookmarks.indexOf(savedSite)
+                bookmarks[index] = savedSite.copy(deleted = "1")
+                val bookmarkFolders = viewState.value?.bookmarkFolders!!
```

Comment:
I see too many `!!` in the new logic (this method and the other ones), which is kind of dangerous and I think can be the root cause of one bug I've reported. can we apply early return or use `let` instead?